### DISCUSSION TITLE
Sharpening the user interface

### DISCUSSION THREAD

**Discussion by ali-ramadhan** [CliMA/Oceananigans.jl#1336] [General]
Thanks to an influx of new users there has been an increase in the number of issues related to the user interface: it's a little clunky and inconsistent in places, should be throwing useful errors to help users in other places, little bugs here and there, etc.

The plan for the next few months is to develop a hydrostatic Boussinesq model running on the sphere which will involve adding new features to Oceananigans. I think it would be good to make sure the user interface continues to improve and remain easy-to-use over the next few months as well (and that mistakes aren't carried forward so much).

Should we list some issues that should be closed/addressed?

**Comment by glwagner** [comment]
The three I'd like to tackle in the near future are:

* https://github.com/CliMA/Oceananigans.jl/issues/890 (Check for 'invalid' boundary conditions)
* https://github.com/CliMA/Oceananigans.jl/issues/1245 (Change default advection scheme and halo size for grids, and add utilities for inferring needed halo sizes?)
* https://github.com/CliMA/Oceananigans.jl/issues/1130 (Derivatives and interpolation of ComputedFields is incorrect on boundaries)

I would also like to address

* https://github.com/CliMA/Oceananigans.jl/issues/1138 (Redesigning `Simulation` and `run!`)

but this is more of a medium-term goal.

**Comment by ali-ramadhan** [comment]
Some issues I'd like to address soon:
* https://github.com/CliMA/Oceananigans.jl/issues/1117
* https://github.com/CliMA/Oceananigans.jl/issues/1119
* https://github.com/CliMA/Oceananigans.jl/issues/1132
* https://github.com/CliMA/Oceananigans.jl/issues/1140
* https://github.com/CliMA/Oceananigans.jl/issues/1256
* https://github.com/CliMA/Oceananigans.jl/issues/1294
